test-framework/junit5-component/src/main/java/io/quarkus/test/component/QuarkusComponentTestExtension.java

```diff
@@ -421,6 +422,7 @@ private void startContainer(ExtensionContext context, Lifecycle testInstanceLife
             ClassLoader tccl = Thread.currentThread().getContextClassLoader();
             SmallRyeConfigBuilder configBuilder = new SmallRyeConfigBuilder().forClassLoader(tccl)
```

Comment:
Couldn't we use https://github.com/quarkusio/quarkus/blob/main/core/runtime/src/main/java/io/quarkus/runtime/configuration/ConfigUtils.java#L76 and let the `Converters` be discovered?

Reply:
We could but I'm not 100% sure it's the best fit for `QuarkusComponentTest`.

For example, we create a new `Config` for each test method. And the discovery itself could be quite expensive. Also the discovered components may require some functionality provided by Quarkus runtime - which might not be available in `QuarkusComponentTest`.

Reply:
That said, we could make it configurable...

Reply:
Done.

Reply:
I see. But the functionality is no different compared to the plain `Consumer<SmallRyeConfigBuilder>` in this PR, is it?

Reply:
No. It was just to let you know there is a way to influence the `SmallRyeConfigBuilder` via `ServiceLoader`, so you may not need a specific API to expose it.

Reply:
I get it, but in the context of `QuarkusComponentTest` we would like to avoid using `ServiceLoader`s.